src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -929,34 +929,6 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     }
 #endif
 
-#if (defined(HOST_ARM64) || defined(HOST_LOONGARCH64)) && !defined(TARGET_APPLE)
-    if (cacheSize == 0)
-    {
-        // We expect to get the L3 cache size for Arm64 but currently expected to be missing that info
```

Comment:
There is a newer copy of this logic below (look for "We expect to get the L3 cache size for Arm64..."). As far as I can tell, the cacheSize computed by this logic is never used - it is always going to be overwritten by newer copy of this logic below.

Reply:
This was introduced by https://github.com/dotnet/runtime/pull/71029/files#diff-3a1328fd045da7e04dc445749425660ae0afd8f352aab38b9dc1f7a6a3b987beR943 . cc @kunalspathak

Reply:
The logic introduced in https://github.com/dotnet/runtime/pull/71029 takes care of machines that did not report L3 cache sizes accurately e.g. Ampere, but after FW update, this should no longer be the case. The dead code was not part of that PR and from my understanding was there to guesstimate a decent cacheSize value (linearly) proportionate to the logical core numbers. @Maoni0 should be able to comment if that logic (or even the one I added) is still relevant or not.

Reply:
> The dead code that is being removed was not part of that PR

The PR that I have linked made the existing code ineffective. Before that PR, the value computed by the code that I am removing was returned by the method.

After that PR, the value computed by the code that I am removing is always thrown away and replaced by a new value computed by the code under `if (cache_level != 3)` condition. This condition is always going to be true when the code that I am removing kicked in.

Reply:
right, because we thought that it is not accurate results returned by reading the L3 cache size value. Makes sense.